projects/storefrontlib/src/cms-components/order-confirmation/components/order-confirmation-thank-you-message/order-confirmation-thank-you-message.component.html

```diff
@@ -1,16 +1,101 @@
-<header class="cx-page-header" *ngIf="(order$ | async) as order">
-  <h1 class="cx-page-title">
-    {{ 'checkoutOrderConfirmation.confirmationOfOrder' | cxTranslate }}
-    {{ order.code }}
-  </h1>
-</header>
+<ng-container *ngIf="(order$ | async) as order">
+  <header class="cx-page-header">
+    <h1 class="cx-page-title">
+      {{ 'checkoutOrderConfirmation.confirmationOfOrder' | cxTranslate }}
+      {{ order.code }}
+    </h1>
+  </header>
 
-<div class="cx-order-confirmation-message">
-  <h2>{{ 'checkoutOrderConfirmation.thankYou' | cxTranslate }}</h2>
-  <p>
-    {{ 'checkoutOrderConfirmation.invoiceHasBeenSentByEmail' | cxTranslate }}
-  </p>
-  <!-- <a href="#" class="btn-link">Print Page</a> -->
-</div>
+  <div class="cx-order-confirmation-message">
+    <h2>{{ 'checkoutOrderConfirmation.thankYou' | cxTranslate }}</h2>
+    <p>
+      {{ 'checkoutOrderConfirmation.invoiceHasBeenSentByEmail' | cxTranslate }}
+    </p>
+    <!-- <a href="#" class="btn-link">Print Page</a> -->
+  </div>
 
-<cx-add-to-home-screen-banner></cx-add-to-home-screen-banner>
+  <div *ngIf="isGuestCustomer">
+    <h2>{{ 'checkoutOrderConfirmation.createAccount' | cxTranslate }}</h2>
+    <p>
+      {{
+        'checkoutOrderConfirmation.createAccountForNext'
+          | cxTranslate: { email: order.paymentInfo.billingAddress.email }
+      }}
+    </p>
+    <div>
+      <form [formGroup]="guestRegisterForm">
```

Comment:
Do you think we should move this to a new component? I think it would clean up this one, both html and ts seem to be bloated with the form logic.